test/MintBurnERC20.test.js

```diff
@@ -0,0 +1,176 @@
+      signatures = await getSignaturesForExecuteTransfer(
+        [mintBurnErc20.address],
+        [otherWallet.address],
+        [amount],
+        [1],
+        batchNonce,
+      );
+    });
+
+    function getExecuteTransferData(tokenAddresses, recipientAddresses, amounts, depositNonces, batchNonce) {
+      const signMessageDefinition = ["address[]", "address[]", "uint256[]", "uint256[]", "uint256", "string"];
+      const signMessageData = [
+        recipientAddresses,
+        tokenAddresses,
+        amounts,
+        depositNonces,
+        batchNonce,
+        "ExecuteBatchedTransfer",
+      ];
+
+      const bytesToSign = ethers.utils.defaultAbiCoder.encode(signMessageDefinition, signMessageData);
+      const signData = ethers.utils.keccak256(bytesToSign);
+      return ethers.utils.arrayify(signData);
+    }
+
+    async function getSignaturesForExecuteTransfer(
+      tokenAddresses,
+      recipientAddresses,
+      amounts,
+      depositNonces,
+      batchNonce,
+    ) {
+      dataToSign = getExecuteTransferData(tokenAddresses, recipientAddresses, amounts, depositNonces, batchNonce);
+      signature1 = await adminWallet.signMessage(dataToSign);
+      signature2 = await relayer1.signMessage(dataToSign);
+      signature3 = await relayer2.signMessage(dataToSign);
+      signature4 = await relayer3.signMessage(dataToSign);
+      signature5 = await relayer5.signMessage(dataToSign);
+      signature6 = await relayer6.signMessage(dataToSign);
+      signature7 = await relayer7.signMessage(dataToSign);
+
+      return [signature1, signature2, signature3, signature4, signature5, signature6, signature7];
+    }
+
+    it("transfer is set as REJECTED when ERC20Safe does not have the minter role", async function () {
+      // check that transfer is set as REJECTED when ERC20Safe does not have the minter role
+      // Get initial balance of the otherWallet
+      const initialBalanceOtherWallet = await mintBurnErc20.balanceOf(otherWallet.address);
+
+      // Execute the transfer
+      await bridge.executeTransfer(
+        [mintBurnErc20.address],
+        [otherWallet.address],
+        [amount],
+        [1],
+        batchNonce,
+        signatures,
+      );
+
+      // Get final balance of the otherWallet
+      const finalBalanceOtherWallet = await mintBurnErc20.balanceOf(otherWallet.address);
+
+      // Assert that the balance hasn't changed
+      expect(initialBalanceOtherWallet).to.equal(finalBalanceOtherWallet);
+
+      const settleBlockCount = await bridge.batchSettleBlockCount();
+      for (let i = 0; i < settleBlockCount; i++) {
+        await network.provider.send("evm_mine");
+      }
+
+      // check that the transfer is set as REJECTED
+      const transfers = await bridge.getStatusesAfterExecution(batchNonce);
+      console.log(transfers);
+      expect(transfers[0]).to.equal(4);
+    });
+    it("transfer is set as Executed when ERC20Safe does have the minter role", async function () {
```

Comment:
Please leave an empty line before starting a new test so it would be easier to read

Reply:
done